autointent/generation/description_generation.py

```diff
@@ -0,0 +1,125 @@
+
+    return intent_utterances
+
+
+async def create_intent_description(
+    client: AsyncOpenAI, intent_name: str | None, utterances: list[str], model_name: str
+) -> str:
+    """
+    Generates a description for a specific intent using an OpenAI model.
+
+    Args:
+        client (AsyncOpenAI): The OpenAI client instance used to communicate with the model.
+        intent_name (str | None): The name of the intent to describe. If None, an empty string will be used.
+        utterances (list[str]): A list of example utterances related to the intent.
+        model_name (str): The identifier of the OpenAI model to use for generating the description.
+
+    Returns:
+        str: The generated description of the intent.
+
+    Raises:
+        ValueError: If the response from the model is not a string or is in an unexpected format.
+    """
+    intent_name = intent_name if intent_name is not None else ""
+    content = PROMPT_DESCRIPTION.format(intent_name=intent_name, user_utterances="\n".join(utterances[:5]))
+    chat_completion = await client.chat.completions.create(
+        messages=[{"role": "user", "content": content}],
+        model=model_name,
+    )
+    result = chat_completion.choices[0].message.content
+
+    if not isinstance(result, str):
+        error_text = f"Unexpected response type: expected str, got {type(result).__name__}"
+        raise TypeError(error_text)
+    return result
+
+
+async def generate_intent_descriptions(
+    client: AsyncOpenAI, intent_utterances: dict[int, list[str]], intents: list[Intent], model_name: str
+) -> list[Intent]:
+    """
+    Generates descriptions for a list of intents using an OpenAI model.
+
+    Args:
+        client (AsyncOpenAI): The OpenAI client used to generate the descriptions.
+        intent_utterances (dict[int, list[str]]): A dictionary mapping intent IDs to their corresponding utterances.
+        intents (list[Intent]): A list of intents to generate descriptions for.
+        model_name (str): The name of the OpenAI model to use for generating descriptions.
+
+    Returns:
+        list[Intent]: The list of intents with updated descriptions.
+    """
+    tasks = []
+    for intent in intents:
+        if intent.description is not None:
+            continue
+        utterances = intent_utterances.get(intent.id, [])
+        task = asyncio.create_task(
+            create_intent_description(
+                client=client,
+                intent_name=intent.name,
+                utterances=utterances,
+                model_name=model_name,
+            )
+        )
+        tasks.append((intent, task))
+
+    descriptions = await asyncio.gather(*(task for _, task in tasks))
+    for (intent, _), description in zip(tasks, descriptions, strict=False):
+        intent.description = description
+    return intents
+
+
+def enhance_dataset_with_descriptions(
+    dataset: Dataset, api_base: str, api_key: str, model_name: str = "gpt-3.5-turbo"
```

Comment:
```suggestion
    dataset: Dataset, api_base: str, api_key: str, model_name: str = "gpt-4o-mini"
```
Она быстрее, лучше дешевле